signalfx/signalflow/computation.py

```diff
@@ -166,7 +166,8 @@ def stream(self):
                 self._process_info_message(message.message)
                 self._batch_count_detected = True
                 yield message
-                if self._current_batch_message:
+                if (self._current_batch_message and
+                        self._current_batch_count == self._expected_batches):
```

Comment:
I think it could be helpful to know more about the intended behavior of out of order info messages instead of propagating state changes that are data message specific. I'm also wondering why `_expected_batches` never resets.

Reply:
right, `_expected_batches` does not change once it's set.